### DISCUSSION TITLE
Does NNCF support power-of-two scaling?

### DISCUSSION THREAD

**Discussion by V0XNIHILI** [openvinotoolkit/nncf#1646] [Q&A] [answered]
For deploying quantized neural networks on custom hardware, power of-two scaling is necessary to avoid the overhead of performing integer divisions in silicon.

With the rich feature-set that NNCF has to offer, I really enjoy using it but it left me wondering whether NNCF also supports power-of-two scaling?

**Comment by AlexKoff88** [comment, marked as answer]
Hi Douwe den Blanken!

Thanks for the interest to NNCF.

NNCF supports log scaling (exponential logarithm). But it can be easily extended to support the power-of-two quantization as well. This is just a matter of how you compute the scale factor internally. Feel free to contribute if you consider this important for you.
Try to search for `logarithm_scale` in the NNCF code for details.
